lib/que/adapters/active_record_with_lock.rb

```diff
@@ -32,17 +45,25 @@ def execute(command, params = [])
         end
       end
 
+      # This method continues looping through the que_jobs table until it either
+      # locks a job successfully or determines that there are no jobs to process.
       def lock_job_with_lock_database(queue, cursor)
-        result = []
         loop do
-          result = Que.execute(:find_job_to_lock, [queue, cursor])
+          observe(duration_metric: FindJobSecondsTotal, labels: { queue: queue }) do
+            locked_job = Que.transaction do
+              job_to_lock = Que.execute(:find_job_to_lock, [queue, cursor])
 
-          break if result.empty?
+              return job_to_lock if job_to_lock.empty?
 
-          cursor = result.first["job_id"]
-          break if pg_try_advisory_lock?(cursor)
+              cursor = job_to_lock.first["job_id"]
+              job_locked = pg_try_advisory_lock?(cursor)
+
+              observe(count_metric: FindJobHitTotal, labels: { queue: queue, job_hit: job_locked })
+              return job_to_lock if job_locked
+            end
+            return locked_job if locked_job
```

Comment:
When is this case hit? Looks like there are only two cases:
> locks a job successfully

on line 64

> or determines that there are no jobs to process.

line 56

Reply:
yeah. This code is not reachable. Will remove this